src/python/grpcio/grpc/_cython/_cygrpc/aio/iomgr/iomgr.pyx.pxi

```diff
@@ -122,11 +124,12 @@ cdef grpc_error* asyncio_socket_listen(grpc_custom_socket* grpc_socket) with gil
     return grpc_error_none()
 
 
-def _asyncio_apply_socket_options(object socket):
-    # TODO(https://github.com/grpc/grpc/issues/20667)
-    # Connects the so_reuse_port option to channel arguments
-    socket.setsockopt(native_socket.SOL_SOCKET, native_socket.SO_REUSEADDR, 1)
-    socket.setsockopt(native_socket.IPPROTO_TCP, native_socket.TCP_NODELAY, True)
+def _asyncio_apply_socket_options(object s, int flags):
+    s.setsockopt(native_socket.SOL_SOCKET, native_socket.SO_REUSEADDR, 1)
+    if platform.system() != 'Windows':
```

Comment:
What would we have to do to get `SO_REUSEADDR` working as well?

Reply:
It should be working with this setup, did I miss anything?

Reply:
My mistake. I see we're unconditionally setting `SO_REUSEADDR`. Is that desirable? How does C++ handle the setting of `SO_REUSEADDR`? Is there any scenario where a user might like to disable this option on Windows?

Reply:
In Core, the `SO_REUSEADDR` is enabled by default for Posix TCP connections, but not other protocols. For Windows, I didn't find a line that manipulates this socket option (maybe it is named differently?).
https://github.com/grpc/grpc/blob/617c43013a68bcdb491dfddeb0eb3e4d3323c53f/src/core/lib/iomgr/tcp_server_utils_posix_common.cc#L175

From what I'm reading, Windows do support `SO_REUSEADDR`: https://docs.microsoft.com/en-us/windows-hardware/drivers/network/so-reuseaddr. It should be fine to supply this flag.

Currently, we only implemented TCP sockets, so we can turn it on by default. If we are going to support UDS, then we will need to change many other components. I'm adding a comments stating this.

Reply:
The point of UDS connections is a good one. We probably want to add a test for that in general.

Reply:
Opened https://github.com/grpc/grpc/issues/21673 to track supporting UDS. It's viable, but needs to tune the iomgr implementation.